extensions/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java

```diff
@@ -192,7 +193,7 @@ public void build(
 
         Collection<AnnotationInstance> paths = index.getAnnotations(ResteasyDotNames.PATH);
 
-        if (paths.isEmpty()) {
+        if (paths.isEmpty() && jaxRsResourceBuildItems.isEmpty()) {
```

Comment:
What bothers me here is that the condition is not exactly right as your resource might not contain a `@Path`.

I wonder if using `GeneratedBeanBuildItem` would be better here as it will get indexed automatically and your JAX-RS resource will be a bean. It should then be automatically considered by the existing code.

You were talking about generated classes so I suppose you have the generated code handy?

@mkouba talking under your control here.

Reply:
> What bothers me here is that the condition is not exactly right as your resource might not contain a @Path.

the "contract" for the JaxrsResourceBuildItem is that it must be a valid resource class which would include the @Path annotation. But I agree we can't enforce it.

> I wonder if using GeneratedBeanBuildItem would be better here as it will get indexed automatically and your JAX-RS resource will be a bean. It should then be automatically considered by the existing code.

That was my initial approach to use the GeneratedBeanBuildItem but that didn't work as the ResteasyServerCommonProcess does not take that into account. Moreover it makes it even more open to not have @Path annotations on it, isn't it?

> You were talking about generated classes so I suppose you have the generated code handy?

Yes, that is available

Reply:
> That was my initial approach to use the GeneratedBeanBuildItem but that didn't work as the ResteasyServerCommonProcess does not take that into account.

Hmmm, I suppose it's because we don't use the `BeanArchiveIndexBuildItem` in `ResteasyServerCommonProcessor`, which would contain these elements.

@mkouba would it make sense switching to the `BeanArchiveIndexBuildItem` considering the `@Path`.. are bean defining annotations so we are sure they are beans.

This way, we could also get the classes added via `GeneratedBeanBuildItem`.

> Moreover it makes it even more open to not have @Path annotations on it, isn't it?

It shouldn't. The idea is to include the class in the index so the normal path can be applied.

Reply:
I think it would make sense to use `BeanArchiveIndexBuildItem` for resources but I'm not quite sure about the other components the `ResteasyServerCommonProcessor` handles.

Anyway,  this `JaxrsResourceBuildItem` approach would not make it possible to inject the resource classes.

Reply:
@mkouba can you elaborate a bit on

> Anyway, this JaxrsResourceBuildItem approach would not make it possible to inject the resource classes.

is there any use case to inject that resource classes themselves? Or did I misunderstand it...

Reply:
I'm sorry I meant to inject a bean into a resource...

Reply:
@mswiderski could you try using `BeanArchiveIndexBuildItem` index to get the resources from (you just need to inject it as a method parameter) and try to use `GeneratedBeanBuildItem` to push your resources to Arc and the index?

I think it should work.

Reply:
Hm, do you register the generated code as a `GeneratedBeanBuildItem`?

Reply:
yes

Reply:
Ok, so what we propose is to use the `BeanArchiveIndexBuildItem` here:
https://github.com/quarkusio/quarkus/blob/master/extensions/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java#L193 so that producing a `GeneratedBeanBuildItem` is the only thing an extension would have to do (no `JaxrsResourceBuildItem` needed).

Reply:
ok, I should produce BeanArchiveIndexBuildItem from the extension which should take care of it already. Let me work on it and update PR

Reply:
or the PR won't be needed anymore if the extension just uses `BeanArchiveIndexBuildItem` and `GeneratedBeanBuildItem`?

Reply:
> ok, I should produce BeanArchiveIndexBuildItem from the extension which should take care of it already.

Not really, just inject the `BeanArchiveIndexBuildItem` in the `ResteasyServerCommonProcessor#build()` method, next to the `CombinedIndexBuildItem` param and use this index instead on line 193 instead.

Reply:
I think it should also be changed here: https://github.com/quarkusio/quarkus/blob/master/extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java#L80 .

Reply:
thanks guys, let me work on it today evening and will update PR as soon as I have it running

Reply:
change applied to use `BeanArchiveIndexBuildItem` instead of additional BuildItem. 

@gsmet about the `ResteasyCommonProcessor`, I haven't made changes there as that part refers to JAX RS providers as far as I understand so it makes less sense to produce providers on  build time.  Or did I miss something?

Reply:
@mswiderski sorry, forgot about this. Yeah, we need to make changes to `ResteasyCommonProcessor`, I'll push a PR for you to check later this week so that we get this one fully sorted out for the next release.